CDP4Common/Poco/ParticipantRole.cs

```diff
@@ -41,31 +40,17 @@ public partial class ParticipantRole
         /// The populate participant permission.
         /// </summary>
         private void PopulateParticipantPermissions()
-        {
-            var pocos = typeof(Thing).QueryAssembly().GetTypes().Where(x => typeof(Thing).QueryIsAssignableFrom(x));
-            foreach (var poco in pocos)
-            {
-                var field = poco.QueryField("DefaultParticipantAccess");
-                if (field == null)
-                {
-                    continue;
-                }
-
-                var accessRight = (ParticipantAccessRightKind)field.GetRawConstantValue();
-                if (accessRight != ParticipantAccessRightKind.NONE)
-                {
-                    continue;
-                }
-
-                ClassKind classkind;
-                Enum.TryParse(poco.Name, out classkind);
-                var personPermission = new ParticipantPermission(Guid.NewGuid(), null, null)
-                {
-                    AccessRight = accessRight,
-                    ObjectClass = classkind
-                };
-                this.ParticipantPermission.Add(personPermission);
-            }
+        {
+            var provider = new DefaultPermissionProvider();
+            provider.GetDefaultTypeNameParticipantPermissions()
+                    .Where(x => x.Value.Equals(ParticipantAccessRightKind.NONE))
+                    .ToList()
+                    .ForEach(x => {
+                        var participantPermission = new ParticipantPermission(Guid.NewGuid(), null, null);
+                        participantPermission.AccessRight = x.Value;
+                        participantPermission.ObjectClass = (ClassKind)Enum.Parse(ClassKind.GetType(), x.Key);
```

Comment:
Maybe store ClassKind.GetType() in a var out of the ForEach loop, but that's up to you.

Reply:
No problem